### PR TITLE
[Datamodules] Update deprecation messages

### PR DESCRIPTION
**Pull request by djdameln**
# Description

Update the deprecation messages related to the datamodules refactor. Remove any references to specific version numbers.

### CODE REVIEW

File: anomalib/config/config.py [hunk 1]
```diff
@@ -241,15 +241,21 @@ def get_configurable_parameters(
     if "metrics" in config.keys():
         # NOTE: Deprecate this after v0.4.0.
         if "adaptive" in config.metrics.threshold.keys():
-            warn("adaptive will be deprecated in favor of method in config.metrics.threshold in v0.4.0.")
+            warn("adaptive will be deprecated in favor of method in config.metrics.threshold in a future release")
```
Comment (samet-akcay) [inline anomalib/config/config.py:244]: `DeprecationWarning`

File: anomalib/config/config.py [hunk 2]
```diff
@@ -241,15 +241,21 @@ def get_configurable_parameters(
...
             config.metrics.threshold.method = "adaptive" if config.metrics.threshold.adaptive else "manual"
         if "image_default" in config.metrics.threshold.keys():
-            warn("image_default will be deprecated in favor of manual_image in config.metrics.threshold in v0.4.0.")
+            warn(
+                "image_default will be deprecated in favor of manual_image in config.metrics.threshold in a future "
```
Comment (samet-akcay) [inline anomalib/config/config.py:248]: `DeprecationWarning`

File: anomalib/config/config.py [hunk 3]
```diff
@@ -241,15 +241,21 @@ def get_configurable_parameters(
...
+                "release."
+            )
             config.metrics.threshold.manual_image = (
                 None if config.metrics.threshold.adaptive else config.metrics.threshold.image_default
             )
         if "pixel_default" in config.metrics.threshold.keys():
-            warn("pixel_default will be deprecated in favor of manual_pixel in config.metrics.threshold in v0.4.0.")
+            warn(
+                "pixel_default will be deprecated in favor of manual_pixel in config.metrics.threshold in a future "
```
Comment (samet-akcay) [inline anomalib/config/config.py:256]: `DeprecationWarning`

File: anomalib/config/config.py [hunk 4]
```diff
@@ -152,14 +152,14 @@ def update_datasets_config(config: Union[DictConfig, ListConfig]) -> Union[DictC
 
     if "clip_length_in_frames" in config.dataset.keys() and config.dataset.clip_length_in_frames > 1:
         warn(
-            "Anomalib's models and visualizer are currently not compatible with video datasets with a clip length > 1.\
-            Custom changes to these modules will be needed to prevent errors and/or unpredictable behaviour."
+            "Anomalib's models and visualizer are currently not compatible with video datasets with a clip length > 1. "
```
[the remark this answers is not in the document]
Comment (djdameln) [reply anomalib/config/config.py:155]: This is not to be deprecated. Just a warning that the model and visualizers do not yet support temporal processing for video.

[the remark this answers is not in the document]
Comment (samet-akcay) [reply anomalib/config/config.py:155]: ah sorry my bad

**Review by samet-akcay — APPROVED** [review]
Thanks

**Pull request merged**